### ISSUE TITLE
Requirement: Implement backpropagation

### ISSUE THREAD

**Issue by Samuel-Nathanson** [Samuel-Nathanson/MachineLearning#8]
Implement backpropagation for training feedforward neural networks. You may choose whether or not
you wish to use the logistic activation function or the hyperbolic tangent activation function for the
hidden layers. The output should be softmax for classification and linear for regression. It is your
choice whether or not you use momentum.
